### PR TITLE
Add embeddings output to model builder

### PR DESCRIPTION
**Pull request by kunal-vaishnavi**
### Description

This PR adds support for outputting the last hidden state in addition to the logits in ONNX models. Users can run their models with ONNX Runtime GenAI and use the generator's `GetOutput` API to obtain the hidden states.

C/C++:
```c
std::unique_ptr<OgaTensor> embeddings = generator->GetOutput("hidden_states");
```

C#:
```csharp
using var embeddings = generator.GetOutput("hidden_states");
```

Java:
```java
Tensor embeddings = generator.getOutput("hidden_states");
```

Python:
```python
embeddings = generator.get_output("hidden_states")
```

### Motivation and Context

In SLMs and LLMs, the last hidden state represents a model's embeddings for a particular input before the language modeling head is applied. Generating embeddings for a model is a popular task. These embeddings can be used for many scenarios such as text classification, sequence labeling, information retrieval using [retrieval-augmented generation (RAG)](https://en.wikipedia.org/wiki/Retrieval-augmented_generation), and more.

This PR helps the following issues:
- https://github.com/microsoft/onnxruntime/issues/20969
- https://github.com/microsoft/onnxruntime-genai/issues/442
- https://github.com/microsoft/onnxruntime-genai/discussions/474
- https://github.com/microsoft/onnxruntime-genai/discussions/713

### CODE REVIEW

File: src/config.h [hunk 1]
```diff
@@ -120,6 +120,7 @@ struct Config {
       } inputs;
 
       struct Outputs {
+        std::string hidden_states{"hidden_states"};
```
[the remark this answers is not in the document]
Comment (kunal-vaishnavi) [reply src/config.h]: Removed `hidden_states` from the config as discussed offline. [This commit](https://github.com/microsoft/onnxruntime-genai/pull/1127/commits/856b8f16425734f31d0c9d85927326ff31f57fea) can be reverted in the future to add it back if needed.

**Review by RyanUnderhill — APPROVED** [review]
